public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -104,6 +105,30 @@ export const QueryAndExpressionsStep: FC<Props> = ({ editingExistingRule }) => {
 
   const emptyQueries = queries.length === 0;
 
+  useEffect(() => {
+    const currentCondition = getValues('condition');
+
+    if (!currentCondition) {
+      return;
+    }
+
+    const error = errorFromSeries(panelData[currentCondition]?.series || []);
+    onDataChange(error?.message || '');
+  }, [panelData, getValues, onDataChange]);
```

Comment:
The reason I'm using this `useEffect` with `panelData` is to be able to check if the condition is valid **once** the data has been updated as a result of `runQueries()`. This happens after either setting a new condition or making changes to the query of the current one.